src/main/java/com/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.bookstore.service.impl;
+
+import com.bookstore.dto.BookDto;
+import com.bookstore.dto.CreateBookRequestDto;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.BookMapper;
+import com.bookstore.model.Book;
+import com.bookstore.repository.BookRepository;
+import com.bookstore.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        bookRepository.save(book);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Book", id)));
+    }
+
+    @Override
+    public List<BookDto> findByTitle(String title) {
+        return bookRepository.findByTitle(title).stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Book", id));
+        Book book = bookMapper.toModel(bookRequestDto);
+        book.setId(id);
+        Book updateBook = bookRepository.save(book);
```

Comment:
By the way, with the `@Transactional` annotation you can update your entities without calling the `save()` method. Read about the dirty-checking mechanism in Hibernate. Note that JPA and Hibernate do not have an `update()` method or anything like that. You can think of the `merge()` method, but its purpose is to add an entity to the session, the update will happen through dirty-checking